commands/MessageCollector.go

```diff
@@ -0,0 +1,42 @@
+package commands
+
+import (
+	"errors"
+	"github.com/bwmarrin/discordgo"
+	"strconv"
+)
+
+/**
+* /collect <message-snowflake-id> <condition-type> <condition-value>
+* Condition type: number, time
+* Condition value: # Messages to collect starting with ID or timestamp to collect till
+*/
+type MessageCollector struct{}
+
+
+func (h MessageCollector) Name() string {
```

Comment:
```suggestion
func (h *MessageCollector) Name() string {
```
In general, if you want to use mutable fields in your struct, its better to implement any interfaces on the struct pointer. As a rule, unless you know for sure that the struct implementing the interface will always be `const`, always implement interfaces on struct pointers.